### PR TITLE
DOC: Add docstring validations for "See Also" section

### LINKED ISSUE #23135: DOC: Validate description of See Also section in docstrings
`scripts/validate_docstrings.py` validates that the content of a docstring follows our standards. There are still some of those standards that the script does not validate, and this gives the users the wrong impression that the docstring is all right, when it's not really the case. All the missing validations are listed in #20298.

One of the standards is that the descriptions in the `See Also` section should start by a capital letter and finish with a period.

For this issue is required:
- Change `scripts/validate_docstings.py` to give an error if the descriptions do not follow to mentioned standard.
- Add tests in `scripts/tests/test_validate_docstrings.py`
- Run the script for all docstrings (i.e. ./scripts/validate_docstings.py) and check in the resulting report how many of the docstrings in pandas have these imports. Add as a comment in this issue the list.

### PR DESCRIPTION
**Pull request by getschomp**
- [x] closes #23135 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: scripts/tests/test_validate_docstrings.py [hunk 1]
```diff
@@ -334,6 +334,33 @@ def method(self, foo=None, bar=None):
...
+        Series.iloc : Return a slice of the elements in the Series,
+            which can also be used to return the first or last n
+        """
+        pass
+
+    def desc_first_letter_lowercase(self):
+        """
+        Return the first 5 elements of the Series.
+
+        see Also
```
Comment (datapythonista) [inline scripts/tests/test_validate_docstrings.py]: I think the lower case is a typo here, right?

File: scripts/tests/test_validate_docstrings.py [hunk 2]
```diff
@@ -564,6 +591,11 @@ def test_bad_generic_functions(self, func):
         assert errors
 
     @pytest.mark.parametrize("klass,func,msgs", [
+        # See Also tests
+        ('BadSeeAlso', 'desc_no_period',
+         ('Missing period at end of description',)),
+        ('BadSeeAlso', 'desc_first_letter_lowercase',
+         ('should be capitalized',)),
```
Comment (datapythonista) [inline scripts/tests/test_validate_docstrings.py]: Can you test that the element reported is the element with the problem? Meaning that `Series.iloc` is ok, and in the error message we want to have that this is for `Series.tail`.

File: scripts/validate_docstrings.py [hunk 3]
```diff
@@ -496,10 +496,18 @@ def validate_one(func_name):
...
+                    errs.append('Missing period at end of description for '
+                                'See Also "{}" reference'.format(rel_name))
+                if not rel_desc[0].isupper():
+                    errs.append('Description should be capitalized for '
+                                'See Also "{}" reference'.format(rel_name))
+            else:
                 errs.append('Missing description for '
                             'See Also "{}" reference'.format(rel_name))
 
+
```
Comment (datapythonista) [inline scripts/validate_docstrings.py]: This blank line is not valid according to PEP-8. If you take a look at the checks, you'll see that travis is red, and if you go to its log, you'll see that this is the reason.

**Review by datapythonista — COMMENTED** [review]
Thanks for the PR. Looks quite good, just couple of comments.

**Comment by getschomp** [comment]
Thanks, @datapythonista, made changes as requested.

**Review by datapythonista — APPROVED** [review]
lgtm. Thanks for the contribution @getschomp 

@WillAyd can you take a look and merge on green if you're happy?

**Comment by getschomp** [comment]
Thanks, @datapythonista, any word @WillAyd?

**Review by WillAyd — APPROVED** [review]

**Pull request merged**